public/app/features/dashboard-scene/settings/version-history/HistorySrv.ts

```diff
@@ -28,15 +29,22 @@ export class HistorySrv {
     return getBackendSrv().get(`api/dashboards/uid/${dashboardUID}/versions`, options);
   }
 
-  getDashboardVersion(uid: string, version: number) {
-    return getBackendSrv().get(`api/dashboards/uid/${uid}/versions/${version}`);
+  getDashboardVersion(dashboardUID: string, version: number) {
+    if (typeof dashboardUID !== 'string' || !isNumber(version)) {
+      return Promise.resolve({});
+    }
```

Comment:
why is this check requited? This should be guaranteed by the API types.

Reply:
In the case the `version` itself is not necessary, at least not anymore, but the `uid` is typed as `any` in `DashboardModel` and there are calls to these functions with it which cause issues.

Reply:
actually, maybe this can be simplified and we can just leave the api deal with wrong uids and versions. I'm looking into it